SmartDeviceLink-iOS/SmartDeviceLink/SDLProtocol.m

```diff
@@ -325,6 +362,10 @@ - (void)destructObjects {
         self.messageRouter = nil;
         self.transport = nil;
         self.protocolDelegate = nil;
+        if (self.heartbeatTimer != nil) {
+            [self.heartbeatTimer cancel];
```

Comment:
No need to cancel here. When it is nil'd, it automatically cancels in `-[SDLTimer dealloc]`.